x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plan/physical/LocalRelation.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.plan.physical;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.xpack.eql.session.EqlSession;
+import org.elasticsearch.xpack.eql.session.Executable;
+import org.elasticsearch.xpack.eql.session.Results;
+import org.elasticsearch.xpack.ql.expression.Attribute;
+import org.elasticsearch.xpack.ql.plan.logical.LogicalPlan;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Collections.emptyList;
+
+public class LocalRelation extends LogicalPlan implements Executable {
+
+    private final Executable executable;
+
+    public LocalRelation(Source source, Executable executable) {
+        super(source, emptyList());
+        this.executable = executable;
+    }
+
+    @Override
+    protected NodeInfo<LocalRelation> info() {
+        return NodeInfo.create(this, LocalRelation::new, executable);
+    }
+
+    @Override
+    public LogicalPlan replaceChildren(List<LogicalPlan> newChildren) {
+        throw new UnsupportedOperationException("this type of node doesn't have any children to replace");
+    }
+
+    public Executable executable() {
+        return executable;
+    }
+
+    @Override
+    public boolean expressionsResolved() {
+        return true;
+    }
+
+    @Override
+    public List<Attribute> output() {
+        return executable.output();
+    }
+
+    @Override
+    public void execute(EqlSession session, ActionListener<Results> listener) {
+        executable.execute(session, listener);
+    }
+
+    @Override
+    public int hashCode() {
+        return executable.hashCode();
```

Comment:
this would NPE if executable is null

Reply:
The NPE cannot occur since:
1. all children are non null as the plan is constructed.
2. if by accident a null child is passed, the base class `Node` checks and throws a proper exception.

Reply:
Thanks for the explanation!